collector/btrfs_linux.go

```diff
@@ -52,34 +56,179 @@ func NewBtrfsCollector(logger log.Logger) (Collector, error) {
 func (c *btrfsCollector) Update(ch chan<- prometheus.Metric) error {
 	stats, err := c.fs.Stats()
 	if err != nil {
-		return fmt.Errorf("failed to retrieve Btrfs stats: %w", err)
+		return fmt.Errorf("failed to retrieve Btrfs stats from procfs: %w", err)
+	}
+
+	ioctlStatsMap, err := c.getIoctlStats()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve Btrfs stats with ioctl: %w", err)
 	}
 
 	for _, s := range stats {
-		c.updateBtrfsStats(ch, s)
+		// match up procfs and ioctl info by filesystem UUID
+		ioctlStats := ioctlStatsMap[strings.Replace(s.UUID, "-", "", -1)]
```

Comment:
Can we be sure there is a entry in ioctlStatsMap for each stats? Otherwise this will panic